### PR TITLE
mlayer-dataset-API

### PR DESCRIPTION
**Pull request by tanvi029**

### CODE REVIEW

File: src/main/java/iudx/catalogue/server/database/Constants.java [hunk 1]
```diff
@@ -141,7 +141,24 @@ public class Constants {
...
+  public static final String GET_MLAYER_INSTANCE_ICON_PATH =
+          "{ \"query\": { \"bool\": { \"minimum_should_match\": 1, \"should\": [$1]}},\"_source\": {\"includes\": [\"icon\",\"name\"] }}";
+  public static final String GET_MLAYER_PROVIDER_DESCRIPTION =
+          "{ \"query\": { \"bool\": { \"minimum_should_match\": 1, \"should\": [$1]}},\"_source\": {\"includes\": [\"id\",\"description\"] }}";
+  public static final String GET_MLAYER_BOOL_PROVIDER =
+      "{\"bool\": {\"must\": [{\"match\": {\"id.keyword\":\"$2\"}},{\"match\":{\"type.keyword\":\"iudx:Provider\"}}]}}";
+  public static final String GET_MLAYER_ALL_DATASETS =
+      "{\"query\":{\"bool\":{\"must\":{\"match\":{\"type.keyword\":\"iudx:ResourceGroup\"}}}},\"_source\":{\"includes\": [\"id\",\"label\",\"accessPolicy\",\"tags\",\"instance\",\"provider\"]}}";
+  public static final String GET_MLAYER_ALL_RESOURCE_DATASETS =
+      "{\"query\":{\"bool\":{\"must\":{\"match\":{\"type.keyword\":\"iudx:Resource\"}}}},\"_source\":{\"includes\": [\"id\",\"resourceGroup\"]},\"size\":980}";
```
Comment (pranavrd) [inline src/main/java/iudx/catalogue/server/database/Constants.java:155]: remove size field

File: src/main/java/iudx/catalogue/server/database/ElasticClient.java [hunk 2]
```diff
@@ -402,7 +414,50 @@ public void onSuccess(Response response) {
...
+                  datasetJson.put("id",record.getString("id")).
+                          put("label",record.getString("label"))
+                          .put("description",record.getString("description"))
+                          .put("accessPolicy",record.getString("accessPolicy"))
+                          .put("instance",record.getString("instance"))
+                          .put("dataSample",record.getJsonObject("dataSample"))
+                          .put("dataSampleFile",record.getJsonArray("dataSampleFile"))
+                          .put("dataQualityFile",record.getJsonArray("dataQualityFile"))
+                          .put("dataDescriptor",record.getJsonObject("dataDescriptor"))
+                          .put("schema",schema);
```
Comment (pranavrd) [inline src/main/java/iudx/catalogue/server/database/ElasticClient.java:453]: Use string constants wherever possible

**Comment by pranavrd** [comment]
ok to test

**Review by kailash — CHANGES_REQUESTED** [review]

**Comment by pranavrd** [comment]
LGTM

**Review by kailash — APPROVED** [review]
LGTM

**Pull request merged**